File: src/main/scala/com/spark1/scala/ParallelizeCollection.scala
package scala.com.spark1.scala

import org.apache.spark.{SparkConf, SparkContext}

/**
  * SortBy || SortByKey
  */
object ParallelizeCollection {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("ParallelizeCollection").setMaster("local")
    val sc = new SparkContext(conf)
    println(sc.parallelize(Array(("Jack",211),("App",112),("Lucy",23),("Zues",100)),1).sortByKey(false).collect().foreach(println(_)))
    println(sc.parallelize(Array(("Jack",211),("App",112),("Lucy",23),("Zues",100)),1).sortBy(_._2,false).collect().foreach(println(_)))
  }

}

File: src/main/scala/com/spark1/java/ParallelizeCollection.java
package scala.com.spark1.java;

import org.apache.spark.SparkConf;
import org.apache.spark.SparkContext;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function2;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created by JACK on 2018/2/15.
 */
public class ParallelizeCollection {

    public static void main(String[] args) {

        SparkConf conf = new SparkConf().setAppName("ParallelizeCollection").setMaster("local");

        JavaSparkContext sc = new JavaSparkContext(conf);

        List<Integer> nums = Arrays.asList(new Integer[]{1,2,3,4,5,6,7,8,9,10});

        JavaRDD<Integer> rdd = sc.parallelize(nums);
        Integer sum = rdd.reduce(((v1, v2) -> v1+v2));
        System.out.println(sum);

    }

}
